src/main/java/net/mcreator/blockly/java/blocks/ProcedureCallBlock.java

```diff
@@ -116,43 +96,84 @@ public class ProcedureCallBlock implements IBlockGenerator {
 				Map<String, Object> dataModel = new HashMap<>();
 				dataModel.put("procedure", procedure.getName());
 				dataModel.put("dependencies", dependencies);
-
-				if (call_at) {
-					dataModel.put("x", xcode);
-					dataModel.put("y", ycode);
-					dataModel.put("z", zcode);
+				if (type.equals("call_procedure")) {
+					dataModel.put("depInputs", depInputs.toArray(DependencyInput[]::new));
 				}
 
-				if (master instanceof BlocklyToJava blocklyToJava
-						&& blocklyToJava.getEditorType() == BlocklyEditorType.COMMAND_ARG) {
-					if (type.equals("old_command")) {
-						master.append(
-								master.getTemplateGenerator().generateFromTemplate("_old_command.java.ftl", dataModel));
-					} else {
-						master.append(master.getTemplateGenerator()
-								.generateFromTemplate("_call_procedure.java.ftl", dataModel));
-					}
+				if (type.equals("old_command")) {
+					master.append(
+							master.getTemplateGenerator().generateFromTemplate("_old_command.java.ftl", dataModel));
 				} else {
-					if (type.equals("call_procedure_at")) {
-						master.append(master.getTemplateGenerator()
-								.generateFromTemplate("_call_procedure_at.java.ftl", dataModel));
-					} else {
-						master.append(master.getTemplateGenerator()
-								.generateFromTemplate("_call_procedure.java.ftl", dataModel));
-					}
+					master.append(
+							master.getTemplateGenerator().generateFromTemplate("_call_procedure.java.ftl", dataModel));
 				}
 			}
 		} else {
+			// No procedure is selected, skip this block
 			master.addCompileNote(new BlocklyCompileNote(BlocklyCompileNote.Type.WARNING,
 					L10N.t("blockly.warnings.call_procedure.empty")));
 		}
 	}
 
+	@Nonnull
+	static List<DependencyInput> mapDependencies(BlocklyToCode master, Element block, List<Dependency> dependencies,

```

Comment:
This method needs JavaDOC explaining purpose, arguments and return value in detail so one can understand what it does, and when one would need to use it

Reply:
When to use a package-private method?

Reply:
Well it is used twice in the package. JavaDOCs should still explain what it does. It is non-trivial method

Reply:
Just not sure how to describe the `master` parameter.

Reply:
Should be good in regards to description for master